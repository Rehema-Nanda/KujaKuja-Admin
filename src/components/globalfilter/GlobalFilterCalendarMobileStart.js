
import React from "react";

import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";
import moment from "moment";
import GlobalFilterCalendarContainer from "./GlobalFilterCalendarContainer";
import CalendarConfig from "./CalendarConfig";

export default class GlobalFilterCalendarMobileStart extends GlobalFilterCalendarContainer {
    constructor(props) {
        super(props);
        this.handleDateChangeStart = this.handleDateChangeStart.bind(this);
        this.calendarContainer = this.calendarContainer.bind(this);
        this.state = {
            selectedDateRangeType: this.getDateRangeType(),
            isMobileView: true,
        };
    }

    handleDateChangeStart(startDate) {
        const { dateEnd, displayMobileFilter } = this.props;

        this.handleDateRangeSelect(startDate, dateEnd, CalendarConfig.IS_START_DATE);
        displayMobileFilter();
    }

    render() {
        const { dateStart, dateEnd, withPortal } = this.props;

        return (

            <div className="date-pickers-holder fadein">
                <DatePicker
                    inline
                    className="date-picker-startdate"
                    withPortal={withPortal}
                    selected={dateStart}
                    selectsStart
                    ref={CalendarConfig.START_DATE_PICKER_REF}
                    startDate={dateStart}
                    endDate={dateEnd}
                    calendarContainer={this.calendarContainer}
                    maxDate={moment.utc().startOf("day")}
                    onChange={this.handleDateChangeStart}
                />
            </div>

        );
    }
}
