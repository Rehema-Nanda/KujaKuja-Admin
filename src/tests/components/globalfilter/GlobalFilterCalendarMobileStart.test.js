import React from "react";
import moment from "moment";
import { shallow } from "enzyme";
import GlobalFilterCalendarMobileStart from "../../../components/globalfilter/GlobalFilterCalendarMobileStart";

describe("<GlobalFilterCalendarMobileStart />", () => {
    const testProps = {
        dateEnd: moment.utc("2018-01-01"),
        setDateStartEnd: jest.fn(),
        displayMobileFilter: jest.fn()
    };
    const wrapper = shallow(<GlobalFilterCalendarMobileStart {...testProps} />);

    it("renders without crashing", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("should call handleDateChangeStart function when the date changes", () => {
        const testFunc = jest.spyOn(wrapper.instance(), "handleDateChangeStart");
        wrapper.instance().forceUpdate();
        const datePicker = wrapper.find("DatePicker");
        datePicker.simulate("change");
        expect(testFunc).toHaveBeenCalled();
    });
});
