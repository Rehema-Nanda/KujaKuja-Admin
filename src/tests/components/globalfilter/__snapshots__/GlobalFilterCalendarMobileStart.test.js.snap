// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalFilterCalendarMobileStart /> renders without crashing 1`] = `
"<div className=\\"date-pickers-holder fadein\\">
  <DatePicker inline={true} className=\\"date-picker-startdate\\" withPortal={false} selected={[undefined]} selectsStart={true} startDate={[undefined]} endDate={{...}} calendarContainer={[Function: bound calendarContainer]} maxDate={{...}} onChange={[Function: bound handleDateChangeStart]} allowSameDay={false} dateFormat=\\"L\\" dateFormatCalendar=\\"MMMM YYYY\\" disabled={false} disabledKeyboardNavigation={false} dropdownMode=\\"scroll\\" onFocus={[Function: onFocus]} onBlur={[Function: onBlur]} onKeyDown={[Function: onKeyDown]} onInputClick={[Function: onInputClick]} onSelect={[Function: onSelect]} onClickOutside={[Function: onClickOutside$$1]} onMonthChange={[Function: onMonthChange]} preventOpenOnFocus={false} onYearChange={[Function: onYearChange]} onInputError={[Function: onInputError]} monthsShown={1} readOnly={false} shouldCloseOnSelect={true} showTimeSelect={false} timeIntervals={30} timeCaption=\\"Time\\" previousMonthButtonLabel=\\"Previous Month\\" nextMonthButtonLabel=\\"Next month\\" renderDayContents={[Function: renderDayContents]} />
</div>"
`;
